### ISSUE TITLE
Qn-Typo

### ISSUE THREAD

**Issue by MothNik** [deepak7376/robustbase#8]
Hi,

I've been looking for a way to compute the Qn scale , when I found a minor mistake in Your file 'robustbase.py'.

In line 75 it says:  

```python
return 2.2219*diff[k-1]
```

According to the R Documentation there was a typo in the very original reference and the prefactor has to be 2.21914 instead of 2.2219.
See https://rdrr.io/cran/robustbase/man/Qn.html.

One can calculate it by using numpy and scipy: 

```python
import numpy as np
from scipy import stats

prefactor = 1 / (np.sqrt(2) * stats.norm.ppf(5/8))
```

which gives 2.219144465985076.

It's just a minor mistake, which doesn't change the result all too much, but getting it right is now a piece of cake.

Kind regards
Nik

**Comment by MothNik** [comment]
There is also a bias correction, that can be easily implemented.
This is adapted from the R source code: https://rdrr.io/cran/robustbase/src/R/qnsn.R

I assume the number of observations is given as n.

```python
# find out if the number of observations is even or odd
even = bool((n + 1) % 2)                                                # returns True for even and False for odd n

# find out if the number of observations is greater than 12
magnitude_n = min(n, 13)

# from that a key is generated
n_key = (even, magnitude_n)

# now the bias correction is calculated
# for even n greater 12
if n_key == (True, 13):

        corr_factor = 3.67561 + ((1.9654 + ((6.987 - (77 / n)) / n)) / n)
        corr_factor = 1 / ((corr_factor / n) + 1)

# for odd n greater 12
elif n_key == (False, 13):

        corr_factor = 1.60188 + ((- 2.1284 - (5.172 / n)) /n)
        corr_factor = 1 / ((corr_factor / n) + 1)

# for n less or equal 12
else:

        # the index of the list reaches from 0 to 10, while n reaches from 2 to 12, so 2 has to be subtracted
        corr_factor = [0.399356, 0.99365, 0.51321, 0.84401, 0.61220,
                               0.85877, 0.66993, 0.87344, 0.72014, 0.88906, 0.75743][n - 2]
```

Running the Python code yields the same results as the R code and these results also match the results in the original reference quite well.

All the credits go to Martin Maechler, I just translated the open source syntax.

**Comment by deepak7376** [comment]
I fixed the prefactor issue, changes it to 2.219144465985076. but i didn't understand the bias correction things. if possible for u to then fix it. Thanks for pointing out error !

**Comment by deepak7376** [comment]
i added the bias correction factor
